python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -409,29 +395,37 @@ def stream_raw_events(self) -> Iterator[DbtCliEventMessage]:
         Returns:
             Iterator[DbtCliEventMessage]: An iterator of events from the dbt CLI process.
         """
-        with self.process.stdout or contextlib.nullcontext():
-            for raw_line in self.process.stdout or []:
-                log: str = raw_line.decode().strip()
-                try:
-                    event = DbtCliEventMessage.from_log(log=log)
-
-                    # Parse the error message from the event, if it exists.
-                    is_error_message = event.raw_event["info"]["level"] == "error"
-                    if is_error_message:
-                        self._error_messages.append(str(event))
-
-                    # Re-emit the logs from dbt CLI process into stdout.
-                    sys.stdout.write(str(event) + "\n")
-                    sys.stdout.flush()
-
-                    yield event
-                except:
-                    # If we can't parse the log, then just emit it as a raw log.
-                    sys.stdout.write(log + "\n")
-                    sys.stdout.flush()
-
-        # Ensure that the dbt CLI process has completed.
-        self._raise_on_error()
+        try:
+            with self.process.stdout or contextlib.nullcontext():
+                for raw_line in self.process.stdout or []:
+                    log: str = raw_line.decode().strip()
+                    try:
+                        event = DbtCliEventMessage.from_log(log=log)
+
+                        # Parse the error message from the event, if it exists.
+                        is_error_message = event.raw_event["info"]["level"] == "error"
+                        if is_error_message:
+                            self._error_messages.append(str(event))
+
+                        # Re-emit the logs from dbt CLI process into stdout.
+                        sys.stdout.write(str(event) + "\n")
+                        sys.stdout.flush()
+
+                        yield event
+                    except:
+                        # If we can't parse the log, then just emit it as a raw log.
+                        sys.stdout.write(log + "\n")
+                        sys.stdout.flush()
+
+            # Ensure that the dbt CLI process has completed.
+            self._raise_on_error()
+        except DagsterExecutionInterruptedError:
+            logger.info(f"Forwarding interrupt signal to dbt command: `{self.dbt_command}`.")
+
+            self.process.send_signal(signal.SIGINT)
+            self.process.wait()
```

Comment:
I think its good to add a `timeout` here

I think its better to let the rest of the graceful exit code run and have an orphaned dbt process (ideally working on exiting) then to block indefinitely here until the `SIGKILL` comes

Reply:
Added a 2s timeout here (chosen arbitrarily)